src/cpp/session/resources/grid/dtviewer.js

```diff
@@ -72,6 +75,48 @@ var activeColumnInfo = {
 // manually adjusted widths of each column
 var manualWidths = [];
 
+// helper for creating a tag with properties + content
+// (created as a string)
+var createTag = function(tag, content, attributes) {
+
+  // if content is an object and attributes is undefined,
+  // treat this as request to create tag with attributes
+  // but no content
+  if (typeof content === 'object' && typeof attributes == 'undefined')
+  {
+    attributes = content;
+    content = '';
+  }
+
+  // ensure attributes is an object
+  attributes = attributes || {};
+
+  // compute inner attributes
+  var parts = [];
+  for (var key in attributes) {
+
+    // extract value
+    var value = attributes[key];
+
+    // join arrays of values
+    if (Object.prototype.toString.call(value) === '[object Array]')
```

Comment:
It's a real shame that this is the only reliable way to discern whether a JavaScript object is an array. :-(

Reply:
I think `Array.isArray()` is now the de-facto way of testing this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray

but it's available in ES5 (not sure if we can assume support for that in all the browsers we expect RStudio Server to run in)